### PR TITLE
[libc][math][c23] Add {getpayload,setpayload,setpayloadsig}f16 C23 math functions

### PR DESCRIPTION
**Pull request by overmighty**
Part of #93566.

### CODE REVIEW

**Comment by overmighty** [comment]
cc @lntue

**Comment by overmighty** [comment]
Rebased and resolved merge conflicts.

**Comment by overmighty** [comment]
Rebased and resolved merge conflicts.

File: libc/src/__support/FPUtil/BasicOperations.h [hunk 1]
```diff
@@ -265,6 +265,48 @@ totalordermag(T x, T y) {
...
+
+  if (!x_bits.is_nan())
+    return T(-1.0);
+
+  return T(x_bits.uintval() & (FPBits::FRACTION_MASK >> 1));
+}
+
+template <bool IsSignaling, typename T>
+LIBC_INLINE cpp::enable_if_t<cpp::is_floating_point_v<T>, bool>
+setpayload(T *res, T pl) {
```
Comment (lntue) [inline libc/src/__support/FPUtil/BasicOperations.h:281]: change to `T& res`

**Review by lntue — COMMENTED** [review]
LGTM with some nits

**Review by lntue — APPROVED** [review]

**Pull request merged**